modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -0,0 +1,610 @@
+                            if (dp_i == 1) {
+                                if (i < ninputs-1) {
+                                    for (int j = 0; j < di1; j++)
+                                        blck[j] = f(blck[j], ptr_i[j]);
+                                } else {
+                                    for (int j = 0; j < di1; j++)
+                                        ptr[j] = saturate_cast<T>(f(blck[j], ptr_i[j]) * scale);
+                                }
+                            } else {
+                                if (i < ninputs-1) {
+                                    for (int j = 0; j < di1; j++, ptr_i += dp_i)
+                                        blck[j] = f(blck[j], *ptr_i);
+                                } else {
+                                    for (int j = 0; j < di1; j++, ptr_i += dp_i)
+                                        ptr[j] = saturate_cast<T>(f(blck[j], *ptr_i) * scale);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // TODO: AutoBuffer for all allocations, this should be called and cached inside finailize()
+    template <typename T, typename Functor>
+    void nary_forward(
+        const Functor& f, T scale,
+        const std::vector<Mat>& inputs, std::vector<Mat>& outputs
+
+//        int ninputs, const char** inp, const int* inp_ndims,
+//        const int** inp_shape, const size_t** inp_step,
+//        char* out, int out_ndims, const int* out_shape, const size_t* out_step
+        )
+    {
+        int ninputs = inputs.size();
+
+        std::vector<const char*> v_inp;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp), [] (const auto& m) { return m.template ptr<const char>(); });
+        const char** inp = v_inp.data();
+
+        std::vector<int> v_inp_dims;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp_dims), [] (const auto& m) { return m.dims; });
+        const int* inp_ndims = v_inp_dims.data();
+
+        std::vector<const int*> v_inp_shape;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp_shape), [] (const auto& m) { return m.size.p; });
+        const int** inp_shape = v_inp_shape.data();
+
+        std::vector<const size_t*> v_inp_step;
+        std::transform(inputs.begin(), inputs.end(), std::back_inserter(v_inp_step), [] (const auto& m) { return m.step.p; });
+        const size_t** inp_step = v_inp_step.data();
+
+        char* out = outputs[0].ptr<char>();
+
+        int out_ndims = outputs[0].dims;
+        const int* out_shape = outputs[0].size.p;
+        const size_t* out_step = outputs[0].step.p;
+
+        int i, max_ndims = out_ndims > 2 ? out_ndims : 2;
+        for(i = 0; i < ninputs; i++)
+            max_ndims = max_ndims > inp_ndims[i] ? max_ndims : inp_ndims[i];
+        size_t* buf = (size_t*)malloc((ninputs+1)*
```

Comment:
We need to simplify(from reader's perspective) allocation of all these buffers somehow

Reply:
Since we are using buf with autobuffer, I cannot think of a way to simplify the allocation but only make detailed and clear comments to explain the buf.